pages/api/posts/index.js

```diff
@@ -0,0 +1,30 @@
+import dbConnect from "@/db/dbConnect";
+import Post from "@/db/models/Post";
+
+export default async function handler(request, response) {
+  await dbConnect();
+
+  if (request.method === "GET") {
+    try {
+      const posts = await Post.find();
+
+      response.status(200).json(posts);
+    } catch (error) {
+      console.log(error);
+      response.status(400).json({ error: error.message });
+    }
+  }
+
+  if (request.method === "POST") {
+    try {
+      const newPost = request.body;
```

Comment:
I would use this endpoint and add the image upload logic in here.

Reply:
I guess it depends. 

One question might be. Do we ever want post submission without images? (maybe?)

Another argument for separating: It might be easier for the students to picture what is going on. Then they can clearly see the data flow. i.e. first i need to upload my image. then get some image data... and then upload the blog post. 

Also... Maybe at some point the logic would be to upload a new image on a change event (rather than submit). Then having the 2 separate API routes would come in handy.